File: prompts.js
// prompts.js
function getDecisionPrompt(session, userInput) {
    const history_count = session.history ? session.history.length : 0;

    let action_rules;
    if (history_count >= 2 && ["분석", "그만", "완료", "그래프", "결과", "리포트"].includes(userInput)) {
        action_rules = "- 'generate_report': If the user requests analysis (e.g., '분석', '그만', '완료').";
    } else {
        action_rules = `- "reset": If the user wants to start over ("다시", "초기화").
- "add_data": If valid child growth data is extracted.
- "ask_for_info": If essential information is still missing.`;
    }

    return `You are a data extractor for a child growth chatbot. Analyze the user's message, extract key information, and decide the next action based on strict rules.

**Current Session Data (Previous entries):**
${JSON.stringify(session, null, 2)}

**User's New Message:**
"${userInput}"

**Extraction Rules:**
- \`sex\`: "남자", "남아", "아들" -> "male"; "여자", "여아", "딸" -> "female".
- \`age_month\`: Convert years/terms to months. (e.g., "3살" -> 36; "두돌" -> 24).
- \`height_cm\`, \`weight_kg\`: If two numbers like "100 15", infer larger is height, smaller is weight.

**Action Decision Rules:**
${action_rules}

**Your Output MUST be a single, valid JSON object with "action" and "data" keys.**
Example (adding data): User: "우리 딸 24개월 85cm 11.5kg" -> Output: {"action": "add_data", "data": {"sex": "female", "age_month": 24, "height_cm": 85, "weight_kg": 11.5}}
Example (analysis): User: "분석해줘" -> Output: {"action": "generate_report", "data": {}}`;
}
module.exports = { getDecisionPrompt };

File: prompts.py
# prompts.py
import json

def get_decision_prompt(session, user_input):
    """사용자 입력과 세션을 기반으로 AI의 행동을 결정하는 프롬프트를 생성합니다."""
    
    history_count = len(session.get('history', []))
    
    # 2개 이상의 데이터가 쌓였고, 사용자가 분석을 요청하면 'generate_report' 규칙만 적용
    if history_count >= 2 and user_input in ["분석", "그만", "완료", "그래프", "결과", "리포트"]:
        action_rules = "- 'generate_report': If the user requests analysis (e.g., '분석', '그만', '완료')."
    else:
        action_rules = """- "reset": If the user wants to start over ("다시", "초기화").
- "add_data": If valid child growth data (`sex`, `age_month`, `height_cm`, `weight_kg`) is extracted.
- "ask_for_info": If essential information is still missing for the next step."""

    # 최종 프롬프트 조합
    prompt = f"""You are a data extractor for a child growth chatbot. Your role is to analyze the user's message, extract key information, and decide the next action based on strict rules.

**Current Session Data (Previous entries):**
{json.dumps(session, indent=2)}

**User's New Message:**
"{user_input}"

**Extraction Rules:**
- `sex`: Extract from "남자", "남아", "아들" as "male"; "여자", "여아", "딸" as "female".
- `age_month`: Convert years and special terms to months. (e.g., "3살", "세살" -> 36; "두돌" -> 24; "100일" -> 3). If only a number is given, assume it's months.
- `height_cm`, `weight_kg`: If two numbers like "100 15" are given, infer the larger is height and the smaller is weight. Extract numbers even if units are present.

**Action Decision Rules:**
{action_rules}

**Your Output MUST be a single, valid JSON object with "action" and "data" keys.**
- Example (adding data): User says "우리 딸 24개월 85cm 11.5kg" -> Output: {{"action": "add_data", "data": {{"sex": "female", "age_month": 24, "height_cm": 85, "weight_kg": 11.5}}}}
- Example (requesting analysis): User says "분석해줘" -> Output: {{"action": "generate_report", "data": {{}}}}
- Example (asking for info): Session has sex, user says "18개월" -> Output: {{"action": "ask_for_info", "data": {{"age_month": 18}}}}
"""
    return prompt